### PR TITLE
Feature/issue 1/UI passcode screen

### PR DESCRIPTION
**Pull request by timofeylyadinsky**
UI for first passcode screen
1. Viewmodel to take information for ui
2. adding dispatcher module for coroutine
3. UI with jetpack compose

### CODE REVIEW

File: app/src/main/java/com/example/newsapp/MainActivity.kt [hunk 1]
```diff
@@ -10,20 +10,24 @@ import androidx.compose.material3.Text
...
 import com.example.newsapp.ui.theme.NewsAppTheme
+import dagger.hilt.android.AndroidEntryPoint
 
+@AndroidEntryPoint
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             NewsAppTheme {
                 // A surface container using the 'background' color from the theme
```
Comment (MaxSkr-iTechArt) [inline app/src/main/java/com/example/newsapp/MainActivity.kt]: you can remove the commented code

File: app/src/main/java/com/example/newsapp/ui/screen/PasscodeScreen.kt [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.example.newsapp.R
+import com.example.newsapp.domain.IsPasscodeRequiredUseCase
+import com.example.newsapp.viewmodel.PasscodeViewModel
+
+
+const val MAX_LENGTH = 4;
```
Comment (MaxSkr-iTechArt) [inline app/src/main/java/com/example/newsapp/ui/screen/PasscodeScreen.kt]: you can make this private I suppose

File: app/src/main/java/com/example/newsapp/viewmodel/PasscodeViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import javax.inject.Inject
+
+@HiltViewModel
+class PasscodeViewModel @Inject constructor(
+    private val isPasscodeRequiredUseCase: IsPasscodeRequiredUseCase,
+    private val savePasscodeUseCase: SavePasscodeUseCase,
+    private val skipPasscodeUseCase: SkipPasscodeUseCase,
+    private val isPasscodeSkipUseCase: IsPasscodeSkipUseCase,
+    private val isPasscodeCorrectUseCase: IsPasscodeCorrectUseCase
+) : ViewModel() {
```
Comment (MaxSkr-iTechArt) [inline app/src/main/java/com/example/newsapp/viewmodel/PasscodeViewModel.kt:29]: you need to introduce a UI state and use it in the composables https://developer.android.com/topic/architecture/ui-layer#expose-ui-state

File: app/src/main/java/com/example/newsapp/ui/screen/PasscodeScreen.kt [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+@Composable
+fun PasscodeScreen(
+    passcodeViewModel: PasscodeViewModel = hiltViewModel()
+) {
```
Comment (MaxSkr-iTechArt) [inline app/src/main/java/com/example/newsapp/ui/screen/PasscodeScreen.kt:37]: consume the UI state introduced in the ViewModel

**Review by MaxSkr-iTechArt — CHANGES_REQUESTED** [review]

**Comment by timofeylyadinsky** [comment]
1. add passcode state to ViewModel and compose

**Comment by timofeylyadinsky** [comment]
Changes
1. edit view model function
2. edit state class
3. refactor composable function and showing block with information from ui state vm

**Review by MaxSkr-iTechArt — APPROVED** [review]

**Pull request merged**